flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -385,7 +386,8 @@ private SchedulerNG createScheduler(
                         initializationTimestamp,
                         getMainThreadExecutor(),
                         fatalErrorHandler,
-                        jobStatusListener);
+                        jobStatusListener,
+                        new NoOpBlocklistHandler());
```

Comment:
The `JobMaster#blocklistHandler` should be passed here.

Reply:
Maybe abstract a `BlocklistOperations` and only expose `addNewBlockedNodes` to scheduler